scripts/scale-test/setup_cluster.py

```diff
@@ -0,0 +1,237 @@
+import json
+import logging
+import subprocess
+from argparse import ArgumentParser, Namespace
+from typing import List
+
+from benchmarks import juju
+from benchmarks.models import Cluster, Unit
+from benchmarks.utils import timeit
+
+APPLICATION = "microk8s-node"
+DEFAULT_CHANNEL = "1.24/stable"
+
+logging.basicConfig(format="[%(levelname)s]: %(message)s", level=logging.INFO)
+
+
+@timeit
+def install_microk8s(model, units: List[Unit], channel=DEFAULT_CHANNEL):
+    configure_proxy(units)
+    reboot_and_wait(model)
+    install_snap(channel)
+    update_etc_hosts(units)
+
+
+def update_etc_hosts(units: List[Unit]):
+    """
+    Add entries in /etc/hosts of all units for each node in the cluster.
+    This is needed as nodes's hostnames need to be resolvable in order for
+    the add-node/join process to work.
+    """
+    logging.info("Adding units hostnames on /etc/hosts")
+    for u in units:
+        cmd = f"echo {u.ip}\t{u.instance_id} >> /etc/hosts"
+        juju.run(cmd, app=APPLICATION).check_returncode()
+
+
+@timeit
+def install_snap(channel: str):
+    logging.info("Installing microk8s on all units")
+    for cmd, check_returncode in [
+        (f"snap install microk8s --classic --channel={channel}", True),
+        ("sudo usermod -a -G microk8s ubuntu", True),
+        ("sudo chown -f -R ubuntu ~/.kube", False),
+        ("sudo newgrp microk8s", True),
+        ("microk8s start", True),
+        ("microk8s status --wait-ready", True),
```

Comment:
This could be a "microk8s status --wait-ready --timeout=<something like 20 minutes>"